pom.xml

```diff
@@ -219,25 +212,57 @@ THE SOFTWARE.
           <targetJdk>1.${java.level}</targetJdk>
         </configuration>
       </plugin>
+      <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-shade-plugin</artifactId>
```

Comment:
Might make sense to submit as a separate PR or to add to the changelog. Dependency shading is a major enhancement and, probably, just another justification for 4.0

Reply:
This was required to make sure that the newly added dependencies did not get dumped in the core classpath. (Same for the CLI JAR in the core PR.)

Reply:
I'll plan on releasing as 4.0 and we'll leave this as proposed.